src/panels/config/zha/zha-device-endpoint-data-table.ts

```diff
@@ -0,0 +1,178 @@
+  @property({ type: Object }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public narrow = false;
+
+  @property({ type: Boolean }) public selectable = false;
+
+  @property({ type: Array }) public deviceEndpoints: ZHADeviceEndpoint[] = [];
+
+  @query("ha-data-table") private _dataTable!: HaDataTable;
+
+  private _deviceEndpoints = memoizeOne(
+    (deviceEndpoints: ZHADeviceEndpoint[]) => {
+      const outputDevices: DeviceEndpointRowData[] = [];
+
+      deviceEndpoints.forEach((deviceEndpoint) => {
+        outputDevices.push({
+          name:
+            deviceEndpoint.device.user_given_name || deviceEndpoint.device.name,
+          model: deviceEndpoint.device.model,
+          manufacturer: deviceEndpoint.device.manufacturer,
+          id: deviceEndpoint.device.ieee + "_" + deviceEndpoint.endpoint_id,
+          ieee: deviceEndpoint.device.ieee,
+          endpoint_id: deviceEndpoint.endpoint_id,
+          entities: deviceEndpoint.entities,
+        });
+      });
+
+      return outputDevices;
+    }
+  );
+
+  private _columns = memoizeOne(
+    (narrow: boolean): DataTableColumnContainer =>
+      narrow
+        ? {
+            name: {
+              title: "Devices",
+              sortable: true,
+              filterable: true,
+              direction: "asc",
+              grows: true,
+              template: (name) => html`
+                <div
+                  class="mdc-data-table__cell table-cell-text"
+                  @click=${this._handleClicked}
+                  style="cursor: pointer;"
+                >
+                  ${name}
+                </div>
+              `,
+            },
+            endpoint_id: {
+              title: "Endpoint",
+              sortable: true,
+              filterable: true,
+            },
+          }
+        : {
+            name: {
+              title: "Name",
+              sortable: true,
+              filterable: true,
+              direction: "asc",
+              grows: true,
+              template: (name) => html`
+                <div
+                  class="mdc-data-table__cell table-cell-text"
+                  @click=${this._handleClicked}
+                  style="cursor: pointer;"
+                >
+                  ${name}
+                </div>
+              `,
+            },
+            endpoint_id: {
+              title: "Endpoint",
+              sortable: true,
+              filterable: true,
+            },
+            entities: {
+              title: "Associated Entities",
+              sortable: false,
+              filterable: false,
+              width: "50%",
+              template: (entities) => html`
+                <div>
+                  ${entities.length
+                    ? entities.map(
+                        (entity) =>
+                          html`<div
+                            class="mdc-data-table__cell table-cell-text"
+                          >
```

Comment:
Why create multiple cells in a cell?

Reply:
<img width="769" alt="image" src="https://user-images.githubusercontent.com/1335687/80987159-a1dccb00-8dff-11ea-956f-bd42a4c7fbdb.png">

It's how I got the associated entities to render correctly like this, including the elided text. Does that make sense?

Reply:
We should not do that, also, what happens when there are 6 entities?

Reply:
We don't have anything where that would be possible yet... if it comes to pass and we have a situation like that I can elide the list too. Do you have any suggestions?

Reply:
Add `style="overflow: hidden; text-overflow: ellipsis;"` instead of the classes.

Reply:
So it is not possible that there are more than 3 entities associated with a groupable device?

Reply:
Updated it to make sure: 
![image](https://user-images.githubusercontent.com/5662298/81736109-5cbb3780-9496-11ea-9ea2-ccd7be9410d8.png)